packages/react-search-ui/src/SearchProvider.js

```diff
@@ -5,35 +5,34 @@ import { SearchDriver } from "@elastic/search-ui";
 import SearchContext from "./SearchContext";
 
 /**
- * The SearchProvider is the glue that connects the SearchDriver to
- * our React App.
+ * The SearchProvider primarily holds a reference to the SearchDriver and
+ * exposes it to the rest of the application in a Context.
  *
- * It "subscribes" to the SearchDriver in order to be notified of state
- * changes. It then syncs that state with its own state and passes that state
- * down to child components in a React Context. It will also pass down "Actions"
- * from the SearchDriver, which allow child components to update state.
- *
- * Any "Container" component placed within this context will have access
+ * Any "Container" component nested within this component will have access
  * to the Driver's state and Actions.
  *
  */
 class SearchProvider extends Component {
   static propTypes = {
-    children: PropTypes.func.isRequired,
+    children: PropTypes.node,
     // Not providing a shape here because the shape matches the shape of
     // SearchDriver. SearchDriver can do it's own parameter validation.
     config: PropTypes.object
   };
 
-  subscription = state => {
-    this.setState(state);
-  };
+  constructor() {
+    super();
+    this.state = {
+      driver: null
+    };
+  }
 
   componentDidMount() {
     const { config } = this.props;
     this.driver = new SearchDriver(config);
```

Comment:
Could you explain why the initialization not inside the constructor?
```js
this.state = {
  driver: new SearchDriver(config)
};
```

Reply:
@m-sureshraj I thought I posted a reply to this, but it looks like I did not, so rewriting it.

This is in CDM because `SearchDriver` has a dependency right now on `window`. So using this with SSR in a project like Gatsby will error out. By moving this to CDM we are basically avoiding the problem by simply not trying to instantiate it during SSR.

Reply:
@m-sureshraj Updated, thanks. I did not move this to hooks, since I was unsure of the behavior of `useEffect` in SSR.